src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -258,23 +268,46 @@ public static DirectoryInfo CreateDirectory(this DirectorySecurity directorySecu
 
         // In the context of a FileStream, the only ACCESS_MASK ACE rights we care about are reading/writing data and the generic read/write rights.
         // See: https://docs.microsoft.com/en-us/windows/win32/secauthz/access-mask
-        private static FileAccess GetFileStreamFileAccess(FileSystemRights rights)
+        private static FileAccess GetFileAccessFromRights(FileSystemRights rights)
         {
             FileAccess access = 0;
-            if ((rights & FileSystemRights.ReadData) != 0 || ((int)rights & Interop.Kernel32.GenericOperations.GENERIC_READ) != 0)
+
+            if ((rights & FileSystemRights.FullControl) != 0 ||
+                (rights & FileSystemRights.Modify) != 0)
+            {
+                return FileAccess.ReadWrite;
+            }
+
+            if ((rights & FileSystemRights.ReadData) != 0 || // Same as ListDirectory
+                (rights & FileSystemRights.ReadExtendedAttributes) != 0 ||
+                (rights & FileSystemRights.ExecuteFile) != 0 || // Same as Traverse
+                (rights & FileSystemRights.ReadAttributes) != 0 ||
+                (rights & FileSystemRights.ReadPermissions) != 0 ||
+                (rights & FileSystemRights.TakeOwnership) != 0 ||
+                (rights & FileSystemRights.Synchronize) != 0 ||
```

Comment:
https://github.com/dotnet/runtime/pull/59389#discussion_r735374636

Reply:
@adamsitnik What do you suggest we do if the user passes only `Synchronize`? Should we throw for not including at least one read or write right?

Reply:
Or should `access` (line 273) stay with the value `0` and let the p/invoke give us back whatever output it generates due to us passing `0`? Note that I tested passing `0` and no failure happens:

https://github.com/dotnet/runtime/pull/59389#discussion_r735371553

Reply:
I see you gave me a detailed answer suggesting what to do with 0:

https://github.com/dotnet/runtime/pull/59389#discussion_r735405550

Reply:
@adamsitnik  so there are 2 concerns in your comment in the old PR: https://github.com/dotnet/runtime/pull/59389#discussion_r735405550

- Do not map `Synchronize` to anything.
- FileAccess == 0 is invalid, handle it correctly.

The suggested code change you gave me in that comment does not do the intended work, and most of the unit tests fail. So I'd rather keep the code as I currently have it here, I think it is much clearer to understand (wouldn't you agree?) but I should address the two specific concerns.